src/components/dialogs/inputs/reactive-capability-curve-table.js

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import IconButton from '@mui/material/IconButton';
+import { useStyles } from '../dialogUtils';
+import Grid from '@mui/material/Grid';
+import DeleteIcon from '@mui/icons-material/Delete';
+import AddIcon from '@mui/icons-material/ControlPoint';
+
+const minimalValue = { p: '', qminP: '', qmaxP: '' };
+
+function getId(value) {
+    return value?.id ? value.id : 'defaultId';
+}
+
+function getNewId() {
+    return Math.random().toString(36).substring(2);
+}
+
+function hasValueCorrectFormat(valueToTest, index) {
+    return !!(
+        valueToTest &&
+        valueToTest[index] &&
+        valueToTest[index].p !== undefined &&
+        valueToTest[index].qminP !== undefined &&
+        valueToTest[index].qmaxP !== undefined
+    );
+}
+
+function enforceMinimumValues(valueToTest) {
+    let returnValue = [];
+    returnValue[0] = hasValueCorrectFormat(valueToTest, 0)
+        ? valueToTest[0]
+        : minimalValue;
+    returnValue[1] = hasValueCorrectFormat(valueToTest, 1)
+        ? valueToTest[1]
+        : minimalValue;
+    if (valueToTest?.length > 2) {
+        for (let i = 2; i < valueToTest.length; i++) {
+            returnValue[i] = hasValueCorrectFormat(valueToTest, i)
+                ? valueToTest[i]
+                : minimalValue;
+        }
+    }
+    return returnValue;
+}
+
+export const useReactiveCapabilityCurveTableValues = ({
+    tableHeadersIds,
+    Field,
+    inputForm,
+    defaultValues, // format : either undefined or [{ p: '', qminP: '', qmaxP: '' }, { p: '', qminP: '', qmaxP: '' }]
+    isReactiveCapabilityCurveOn,
+    disabled = false,
+}) => {
+    const classes = useStyles();
+
+    // Values sent back to the parent component.
+    const [values, setValues] = useState(enforceMinimumValues(defaultValues));
+
+    // Values displayed in the form. When updated, they also updates "values".
+    const [displayedValues, setDisplayedValues] = useState(
```

Comment:
I don't know, we should avoid those two states which save the exact same thing...
I know it's a kind of a fix for the flickering problem, but seems a bad fix... I will think to another solution